src/mqueryfront/src/QueryPage.js

```diff
@@ -77,127 +98,111 @@ class QueryPage extends Component {
+                "?offset=" +
+                OFFSET +
+                "&limit=" +
+                LIMIT
+        );
+        return response ? response.data : {};
+    }
 
-        axios
-            .get(
-                API_URL +
-                    "/matches/" +
-                    this.state.qhash +
-                    "?offset=" +
-                    OFFSET +
-                    "&limit=" +
-                    LIMIT
-            )
-            .then((response) => {
-                const { job, matches } = response.data;
+    collapsePane() {
+        this.setState((prevState) => ({
+            collapsed: !prevState.collapsed,
+        }));
+    }
 
+    async submitQuery(method, priority) {
+        try {
+            let response = await axios.post(API_URL + "/query", {
+                raw_yara: this.state.rawYara,
+                method: method,
+                priority: priority,
+                taint: this.state.selectedTaint,
+            });
+            if (method === "query") {
+                this.props.history.push("/query/" + response.data.query_hash);
+            } else if (method === "parse") {
                 this.setState({
-                    job: job,
-                    matches: matches,
+                    queryPlan: response.data,
+                    queryError: null,
                 });
-                const isDone = isStatusFinished(job.status);
-                if (isDone) {
-                    return;
-                }
-                this.timeout = setTimeout(() => this.loadJob(), 1000);
+            }
+        } catch (error) {
+            this.setState({
+                queryError: error.response
+                    ? error.response.data.detail
+                    : error.toString(),
+                queryPlan: null,
             });
+        }
     }
 
-    callbackResultsActivePage = (pageNumber) => {
-        this.setState({ activePage: pageNumber }, () => {
-            this.loadMatches();
-        });
-    };
-
-    loadMatches() {
-        const LIMIT = 20;
-        let OFFSET = (this.state.activePage - 1) * 20;
-        axios
-            .get(
-                API_URL +
-                    "/matches/" +
-                    this.state.qhash +
-                    "?offset=" +
-                    OFFSET +
-                    "&limit=" +
-                    LIMIT
-            )
-            .then((response) => {
-                this.setState({
-                    matches: response.data.matches,
-                });
-            });
+    get parsedError() {
+        if (this.state.queryError) {
+            // Dirty hack to parse error lines from the error message
+            // Error format: "Error at 4.2-7:" or  "Error at 5.1:"
+            let parsedError = this.state.queryError.match(
+                /Error at (\d+).(\d+)-?(\d+)?: (.*)/
+            );
+            if (parsedError) return parsedError;
+        }
+        return [];
     }
```

Comment:
is this right to use getter here? the object is manipulated and it doesn't seem like it is exclusively bound to "queryError" which is used on other places. But hey! I am not a JS coder so I dont know more than you do

Reply:
Actually I was also in doubt how this `parsedError` should be implemented. I've just lifted it from QueryField along with query submission method and placed it in getter, because its value seemed to be directly derived from queryError message. But maybe this could be placed in QueryMonaco because that regex has strong dependency with `setError` arguments. Such design decisions are always pain in the neck :/

Reply:
there is no right answer, flip a coin or let `random.choose` to decide :P